### ISSUE TITLE
[Data API] Shows Endpoint does not return schedule in 24 hour format

### ISSUE THREAD

**Issue by neosonic2** [netmix/radio-station#322]
**Describe the bug**
Accessing a list of shows via the /wp-json/radio/shows JSON API endpoint returns show schedule time data in a 12-hour time format rather than in a 24-hour time format, but does not provide AM or PPM context, nor does it provide information as to whether the show time is an encore presentation.

For example, we have a show that airs Saturdays from 7-9 PM and then Sundays from 2-4 AM as an encore presentation. A partial representation of the show's object in the API endpoint returns this data:

```{"id":3428,"name":"ABC Oldies with the Real Jeff Moulton","schedule":[{"day":"Sunday","start":"02:00","end":"04:00"},{"day":"Saturday","start":"07:00","end":"09:00"}]}```

Notice the Saturday start and end times given as 07:00 and 09:00 respectively, when they should be listed as 19:00 and 21:00, respectively.

**To Reproduce**
1. Go to https://www.rochesterfreeradio.com/wp-json/radio/shows for a working example of this problem.
2. 2. Look at the "schedule" object of any show. Notice there is no way to tell whether the show airs in the morning or in the evening, or whether the show time is an encore presentation.

**Expected behavior**
Using the above example, the returned JSON should look like this:

```{"id":3428,"name":"ABC Oldies with the Real Jeff Moulton","schedule":[{"day":"Sunday","start":"02:00","end":"04:00"},{"day":"Saturday","start":"19:00","end":"21:00"}]}```

Additionally, an identifier for an encore presentation should be present in the Sunday schedule object a swell.

Our Website is running the latest versions of WordPress and the Radio Station plugin and show times display properly on the front end show schedule page and on the backend administration interface. It is only the JSON API that returns incorrect information.

**Comment by majick777** [comment]
@neosonic2 You are quite right, good pickup, these should be in 24 hour format. Funny, I thought I had fixed this in the last release, but perhaps it was just a similar problem elsewhere in the data API. I've just pushed an update to the `develop` branch that should fix this issue. Seems like the wrong time conversion function was being used.
https://netmix.com/radio-station/docs/FAQ/#how-do-i-install-the-latest-development-version-for-testing

**Comment by neosonic2** [comment]
Thank you for correcting the time format of the shows API endpoint; I just installed the latest development version of the plugin and I can confirm that times are correctly shown in the 24-hour format.

However, the other issue mentioned in my original report still exists, in that no encore designation is displayed when looking at the shows API endpoint like it is on the schedule API endpoint. Is it possible to update the shows endpoint to display an "encore" boolean value perhaps inside the "schedule" object along side "day", "start", and "end"?

**Comment by majick777** [comment]
@neosonic2 Sorry for the oversight, I missed reading that part completely... okay, I've just added the encore boolean for shift to the latest development version.

**Comment by neosonic2** [comment]
Hi! Thanks for making this change; I just installed the latest development version and can see the new "encore" property you added. One small nitpick though if that's OK: the schedule API endpoint uses boolean true/false values for the encore property but it looks like the shows endpoint uses 0/1. I know they're the same thing technically but I figured consistency is important LOL. Would it be possible to switch the shows endpoint to use true/false like the schedule endpoint as well?
